controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -75,11 +75,6 @@ func NewAdapter(snapshotEnvironmentBinding *applicationapiv1alpha1.SnapshotEnvir
 // EnsureIntegrationTestPipelineForScenarioExists is an operation that will ensure that the Integration test pipeline
 // associated with the Snapshot and the SnapshotEnvironmentBinding's IntegrationTestScenarios exist.
 func (a *Adapter) EnsureIntegrationTestPipelineForScenarioExists() (controller.OperationResult, error) {
-	if gitops.HaveAppStudioTestsFinished(a.snapshot) {
```

Comment:
This is my fix for the deadlock, I couldn't figure out what's the purpose of this gating.

Reply:
maybe because why would we want to create Integration PLRs for a snapshot that has already finished its testing?

Reply:
> maybe because why would we want to create Integration PLRs for a snapshot that has already finished its testing?

I'm re-running old tests in GH in order to figure out if it's regression, or race condition, or bug caused by something else, if new PRs are failing:D